_posts/2019-04-09-spring-security-in-action.md

```diff
@@ -0,0 +1,224 @@
+---
+layout: post
+title:  "Spring Security In Action"
+date:   2019-04-09 23:31:07 +0800
+tags: [JAVA, Spring-Security]
+categories: [blog]
+author: Jakob He
+
+---
+
+## **What is Spring-Security?**
+Spring-Security is a highly customizable authentication and access-control framework for java applications. Especially for spring based applications. This is widely used in Java developing field. 
+
+## **Why do we need to use Spring-Security?**
+Spring-Security is highly integrated with the most popular framework Spring-Boot. It supports both Authentication and Authorization, which are also the most popular ways to deal with the security issues between server and client end. Like all Spring based projects, the real power of Spring-Security is found in how easily it can be extended to meet customer requirements.
+
+## **Spring-Security in Action**
+In this example we will go through a very basic Spring-Security application. There are four important classes to be introduced (HttpSecurity, WebSecurityConfigurer, UserDetailsService, AuthenticationManager). And the final application will cover following features. 
+1. username and password verification
+2. role control
+3. token distribution - using JWT
+4. token verification
+5. password crypto
+
+> ![spring-security-scope.jpg](integration-blog/assets/2019-04-09-spring-security-in-action/spring-security-scope.jpg)
```

Comment:
Cannot see the picture

Reply:
Seeing from other article this link will be visible after deploy, pictures are in assets folder.